channeldb/reports.go

```diff
@@ -0,0 +1,323 @@
+package channeldb
+
+import (
+	"bytes"
+	"errors"
+	"io"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+	"github.com/lightningnetwork/lnd/channeldb/kvdb"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+var (
+	// closeSummaryBucket is a top level bucket which holds additional
+	// information about channel closes. It nests channels by chainhash
+	// and channel point.
+	// [closeSummaryBucket]
+	//	[chainHashBucket]
+	//		[channelBucket]
+	//			[resolversBucket]
+	closeSummaryBucket = []byte("close-summaries")
+
+	// resolversBucket holds the outcome of a channel's resolvers. It is
+	// nested under a channel and chainhash bucket in the close summaries
+	// bucket.
+	resolversBucket = []byte("resolvers-bucket")
+)
+
+var (
+	// ErrNoChainHashBucket is returned when we have not created a bucket
+	// for the current chain hash.
+	ErrNoChainHashBucket = errors.New("no chain hash bucket")
+
+	// ErrNoChannelSummaries is returned when a channel is not found in the
+	// chain hash bucket.
+	ErrNoChannelSummaries = errors.New("channel bucket not found")
+
+	amountType    tlv.Type = 1
+	resolverType  tlv.Type = 2
+	outcomeType   tlv.Type = 3
+	spendTxIDType tlv.Type = 4
+)
+
+// ResolverType indicates the type of resolver that was resolved on chain.
+type ResolverType uint8
+
+const (
+	// ResolverTypeUnknown is a placeholder for unknown resolver types.
+	ResolverTypeUnknown ResolverType = 0
+)
+
+// ResolverOutcome indicates the outcome for the resolver that that the contract
+// court reached. This state is not necessarily final, since htlcs on our own
+// commitment are resolved across two resolvers.
+type ResolverOutcome uint8
+
+const (
+	// ResolverOutcomeUnknown is a placeholder for unknown resolver
+	// outcomes.
+	ResolverOutcomeUnknown ResolverOutcome = 0
```

Comment:
I noticed it isn't actually used anywhere, also not in the full diff. Is this to prevent giving meaning to the zero value?

Reply:
The const values don't use `iota`, so it could be removed?